blockstore/autobatch.go

```diff
@@ -162,6 +162,8 @@ func (bs *AutobatchBlockstore) Get(ctx context.Context, c cid.Cid) (block.Block,
 		return blk, err
 	}
 
+	bs.stateLock.Lock()
+	defer bs.stateLock.Unlock()
```

Comment:
This will hold the lock around the Get at the bottom. We probably don't want that.

Reply:
Honestly, if we're getting here, something kinda weird is happening. I'd rather just hold the lock and be slow.